jdbc-core/src/main/java/com/salesforce/datacloud/jdbc/core/SslProperties.java

```diff
@@ -0,0 +1,349 @@
+            builder.clientKeyPathValue(clientKeyPath);
+
+            // Either ca_cert or Trust store can be present,not both
+            if (props.containsKey(SSL_CA_CERT_PATH)) {
+                String caCertPath = takeOptional(props, SSL_CA_CERT_PATH).orElse(null);
+                validateFilePath(SSL_CA_CERT_PATH, caCertPath);
+                builder.caCertPathValue(caCertPath);
+
+            } else if (props.containsKey(SSL_TRUSTSTORE_PATH)) {
+                String truststorePath = takeOptional(props, SSL_TRUSTSTORE_PATH).orElse(null);
+                validateFilePath(SSL_TRUSTSTORE_PATH, truststorePath);
+
+                builder.truststorePathValue(truststorePath);
+                builder.truststorePasswordValue(
+                        takeOptional(props, SSL_TRUSTSTORE_PASSWORD).orElse(null));
+                builder.truststoreTypeValue(
+                        takeOptional(props, SSL_TRUSTSTORE_TYPE).orElse(DEFAULT_TRUSTSTORE_TYPE));
+            }
+        } else if (props.containsKey(SSL_CLIENT_CERT_PATH) || props.containsKey(SSL_CLIENT_KEY_PATH)) {
+            // Can't have only one of cert or key
+            if (props.containsKey(SSL_CLIENT_CERT_PATH)) {
+                throw new SQLException(
+                        "Client certificate provided but private key is missing. "
+                                + "Both ssl.client.certPath and ssl.client.keyPath are required for mutual TLS.",
+                        "28000");
+            } else {
+                throw new SQLException(
+                        "Client private key provided but certificate is missing. "
+                                + "Both ssl.client.certPath and ssl.client.keyPath are required for mutual TLS.",
+                        "28000");
+            }
+        } else if (props.containsKey(SSL_CA_CERT_PATH) || props.containsKey(SSL_TRUSTSTORE_PATH)) {
+            if (props.containsKey(SSL_CA_CERT_PATH) && props.containsKey(SSL_TRUSTSTORE_PATH)) {
+                throw new SQLException("Cannot specify both ssl.ca.certPath and ssl.truststore.path. ", "HY000");
+            }
+            builder.sslMode(SslMode.ONE_SIDED_TLS);
+
+            if (props.containsKey(SSL_CA_CERT_PATH)) {
+                String caCertPath = takeOptional(props, SSL_CA_CERT_PATH).orElse(null);
+                validateFilePath(SSL_CA_CERT_PATH, caCertPath);
+                builder.caCertPathValue(caCertPath);
+            } else {
+                String truststorePath = takeOptional(props, SSL_TRUSTSTORE_PATH).orElse(null);
+                validateFilePath(SSL_TRUSTSTORE_PATH, truststorePath);
+
+                builder.truststorePathValue(truststorePath);
+                builder.truststorePasswordValue(
+                        takeOptional(props, SSL_TRUSTSTORE_PASSWORD).orElse(null));
+                builder.truststoreTypeValue(
+                        takeOptional(props, SSL_TRUSTSTORE_TYPE).orElse(DEFAULT_TRUSTSTORE_TYPE));
```

Comment:
afaict, if the user specifies only `ssl.truststore.password` but not `ssl.truststore.path`, you would silently accept and ignore the `ssl.truststore.password`.

You are accepting it here, because you are using `takeOptional`, but you didn't check if the user actually also provided a path. And if no path is provided, the code in `createTrustManagerFactory` will ignore the password.

Easiest solution is probably to change the `} else {` to an `} else if (props.containsKey(SSL_TRUSTSTORE_PATH)) {`.